openhands/runtime/utils/file_viewer.py

```diff
@@ -0,0 +1,146 @@
+"""
+Utility module for generating file viewer HTML content.
+"""
+
+import base64
+import mimetypes
+import os
+
+
+def generate_file_viewer_html(file_path: str) -> str:
+    """
+    Generate HTML content for viewing different file types.
+
+    Args:
+        file_path: The absolute path to the file
+
+    Returns:
+        str: HTML content for viewing the file
+
+    Raises:
+        ValueError: If the file extension is not supported
+    """
+    file_extension = os.path.splitext(file_path)[1].lower()
+    file_name = os.path.basename(file_path)
+
+    # Define supported file extensions
+    supported_extensions = [
+        '.pdf',
+        '.png',
+        '.jpg',
+        '.jpeg',
+        '.gif',
+    ]
+
+    # Check if the file extension is supported
+    if file_extension not in supported_extensions:
```

Comment:
@adityasoni9998 I remember you found out that files downloaded by browsergym always miss extensions in their filenames, right?

Maybe we should handle that case here as well and try to guess the file type?

Reply:
Anyways this is non-blocking

Reply:
Yeah, i think we can do that on the browsergym side of work